### PR TITLE
Restore Query::one interface

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️/❌

### CODE REVIEW

File: src/Query/QueryInterface.php [hunk 1]
```diff
@@ -249,10 +249,10 @@ public function getWithQueries(): array;
      * @throws InvalidConfigException
      * @throws Throwable
      *
-     * @return array|null The first row (in terms of an array) of the query result. It returns `null` if the query
-     * results in nothing.
+     * @return array|object|null The first row (as array or object) of the query result.
+     * It returns `null` if the query results in nothing.
      */
-    public function one(): array|null;
+    public function one(): array|object|null;
```
Comment (vjik) [inline src/Query/QueryInterface.php:255]: Should we make same change in `all()` method and in `Query` class?

**Review by Tigrov — COMMENTED** [review]
`BatchQueryResult::$value` and related methods also should have the type `array|object|null`
https://github.com/yiisoft/db/blob/ef81e8b7749525030dba95060b71a9f490fe2784/src/Query/BatchQueryResult.php#L43

File: src/Query/QueryInterface.php (hunk 1 above)
Comment (darkdef) [reply src/Query/QueryInterface.php:256]: fixed

**Comment by darkdef** [comment]
> `BatchQueryResult::$value` and related methods also should have the type `array|object|null`

It's maybe incorrect for query with one column as scalar type.

**Comment by Tigrov** [comment]
> > `BatchQueryResult::$value` and related methods also should have the type `array|object|null`
> 
> It's maybe incorrect for query with one column as scalar type.

This should be an array ($row) or array of arrays ($rows) or null, but seems this is not related with the PR.

**Review by Tigrov — CHANGES_REQUESTED** [review]
Needs line to changelog

File: src/Query/QueryInterface.php [hunk 2]
```diff
@@ -50,7 +50,8 @@ public function addParams(array $params): static;
      * @throws InvalidConfigException
      * @throws Throwable
      *
-     * @return array[] The query results. If the query results in nothing, it returns an empty array.
+     * @return array The query results. If the query results in nothing, it returns an empty array.
+     * @psalm-return array[]|object[]
```
Comment (vjik) [inline src/Query/QueryInterface.php:54]:
```suggestion
     * @return array[]|object[] The query results. If the query results in nothing, it returns an empty array.
     * @psalm-return list<array|object>
```

File: src/Command/CommandInterface.php [hunk 3]
```diff
@@ -631,10 +631,10 @@ public function queryColumn(): array;
      * @throws Exception
      * @throws Throwable If execution failed.
      *
-     * @return array|null The first row (in terms of an array) of the query result. Null if the query
+     * @return array|object|null The first row (in terms of an array) of the query result. Null if the query
```
Comment (vjik) [inline src/Command/CommandInterface.php:634]:
```suggestion
     * @return array|object|null The first row (as array or object) of the query result. Null if the query
```

**Review by vjik — CHANGES_REQUESTED** [review]
Need also fix type of `CommandInterface::queryAll()`

**Comment by vjik** [comment]
> Needs line to changelog

And fill `UPGRADE.md` too

**Review by samdark — APPROVED** [review]

**Review by Tigrov — APPROVED** [review]

**Comment by Tigrov** [comment]
> Do we need also fix type of `CommandInterface::queryAll()`?

Currently it looks excessive and needs refactoring `DbArrayHelper` for supporting objects but it looks redundant.

**Review by vjik — APPROVED** [review]

**Pull request merged**